File: solutions/ocaml/126-meeting-rooms.ml
(*
  Problem 126: Meeting Rooms (LeetCode 252)
  Difficulty: Easy
  Language: OCaml
*)
let can_attend_meetings intervals =
  let sorted = List.sort (fun (a,_) (b,_) -> compare a b) intervals in
  let rec check = function
    | [] | [_] -> true
    | (_, e) :: ((s, _) :: _ as rest) -> e <= s && check rest
  in check sorted

File: solutions/rust/126-meeting-rooms.rs
/**
 * Problem 126: Meeting Rooms (LeetCode 252)
 * Difficulty: Easy
 * Language: Rust
 */
fn can_attend_meetings(intervals: &mut Vec<(i32, i32)>) -> bool {
    intervals.sort();
    intervals.windows(2).all(|w| w[0].1 <= w[1].0)
}
